core/keyserver/keyserver.go

```diff
@@ -636,36 +635,30 @@ func (s *Server) BatchQueueUserUpdate(ctx context.Context, in *pb.BatchQueueUser
 	// - Correct key formats.
 	_, tdone := monitoring.StartSpan(ctx, "BatchQueueUserUpdate.verify")
 	updates := make(chan *pb.EntryUpdate)
-	errors := make(chan error)
 	go func() {
 		defer close(updates)
 		for _, u := range in.Updates {
 			updates <- u
 		}
 	}()
-	go func() {
```

Comment:
What was the race in this block? If there is none, can this be in a separate PR?

Reply:
The race in this block was that `Wait()` was being called concurrent to `Add()`

Reply:
Doesn't look like it. All `Add`s were done synchronously, strictly before the deferred `Wait` was called.

Reply:
Sorry - I was talking about the race condition in core/client/batch_get_and_verify.go

The race condition in this block was `err = validateEntryUpdate`. The fix: `err := validateEntryUpdate`

Reply:
Oh, this is a good one :)

Well the refactoring is a bit out of scope then. But LGTM, as I don't want to block you.